### PR TITLE
fix: address issues in webconfig

### PR DESCRIPTION
**Pull request by andrewrynhard**
Fixes TLS and invalide redirects.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

File: internal/pkg/webconfig/webconfig.go [hunk 1]
```diff
@@ -104,14 +102,6 @@ func (s *Server) Stop() {
 
 // ServeHTTP implements http.Handler.
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if r.URL.Scheme != "https" {
-		r.URL.Scheme = "https"
-
-		http.Redirect(w, r, r.URL.String(), http.StatusTemporaryRedirect)
-
-		return
-	}
```
Comment (andrewrynhard) [inline internal/pkg/webconfig/webconfig.go:113]: Kept getting invalid redirect errors. Took another approach.

File: internal/pkg/webconfig/webconfig.go [hunk 2]
```diff
@@ -217,7 +217,7 @@ func (s *Server) Run(ctx context.Context) ([]byte, error) {
 	}
 
 	go func() {
-		if err = s.svr.ListenAndServe(); err != nil {
+		if err = http.ListenAndServe(":80", http.HandlerFunc(redirect)); err != nil {
```
Comment (andrewrynhard) [inline internal/pkg/webconfig/webconfig.go:220]: Kept it simple and just setup a listener for port 80 that knows how to hand redirects.

File: internal/pkg/webconfig/webconfig.go [hunk 3]
```diff
@@ -232,12 +232,14 @@ func (s *Server) Run(ctx context.Context) ([]byte, error) {
...
-	fmt.Println("Webconfig started.  You may POST a configuration to:")
+	s.logger.Println("Webconfig started. You may POST a configuration to:")
 
 	for _, ip := range ips {
-		fmt.Printf("\t%s\n", ip.String())
+		s.logger.Printf("\thttps://%s\n", ip.String())
 	}
 
+	s.logger.Println("You may also visit any of the above in your browser.")
+
```
Comment (andrewrynhard) [inline internal/pkg/webconfig/webconfig.go:242]: I wasn't seeing this printed out in the console on a real machine. The logger is configured to write to the correct place regardless of which mode we are running.

File: internal/pkg/webconfig/webconfig.go [hunk 4]
```diff
@@ -247,49 +249,33 @@ func (s *Server) Run(ctx context.Context) ([]byte, error) {
...
+	pool.AppendCertsFromPEM(ca.CrtPEM)
 
-	// We need a broader set of ciphers to interact with browsers.
-	tlsConfig.CipherSuites = make([]uint16, len(tls.CipherSuites()))
-	for _, s := range tls.CipherSuites() {
-		tlsConfig.CipherSuites = append(tlsConfig.CipherSuites, s.ID)
+	tlsConfig := &tls.Config{
+		RootCAs:      pool,
+		MinVersion:   tls.VersionTLS12,
+		Certificates: []tls.Certificate{*keypair.Certificate},
```
Comment (andrewrynhard) [inline internal/pkg/webconfig/webconfig.go:278]: Basically, moves to a simple RSA PKI setup.

File: internal/pkg/webconfig/webconfig.go (hunk 3 above)
Comment (Ulexus) [reply internal/pkg/webconfig/webconfig.go:242]: Curious; I used the fmt method because _I_ wasn't seeing the logs with the _logger_.,

**Review by Ulexus — APPROVED** [review]

File: internal/pkg/webconfig/webconfig.go (hunk 3 above)
Comment (andrewrynhard) [reply internal/pkg/webconfig/webconfig.go:242]: Hmm. Well that is strange. Maybe something broken specifically around the logger and container mode. Regardless, I think this is the intended way, and it should work. Sounds like a bug outside of this.

**Pull request merged**